From 1e71053b0be82d2dea54b15bd181e67464c913c3 Mon Sep 17 00:00:00 2001
From: NelloKudo <marshnelloosu@gmail.com>
Date: Fri, 22 May 2026 01:48:35 +0200
Subject: [PATCH] HACK: mfsrcsnk: Enable winedmo backend with WINE_USE_WINEDMO.

---
 dlls/mfsrcsnk/media_source.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dlls/mfsrcsnk/media_source.c b/dlls/mfsrcsnk/media_source.c
index c288383b391..3c2ffe93dc6 100644
--- a/dlls/mfsrcsnk/media_source.c
+++ b/dlls/mfsrcsnk/media_source.c
@@ -2027,6 +2027,10 @@ static BOOL use_gst_byte_stream_handler(void)
 {
     BOOL result;
     DWORD size = sizeof(result);
+    const char *dmocheck = getenv("WINE_USE_WINEDMO");
+
+    if (dmocheck && atoi(dmocheck))
+        return FALSE;
 
     /* @@ Wine registry key: HKCU\Software\Wine\MediaFoundation */
     if (!RegGetValueW( HKEY_CURRENT_USER, L"Software\\Wine\\MediaFoundation", L"DisableGstByteStreamHandler",
-- 
2.54.0

